### DISCUSSION TITLE
Import dataset from mounted share

### DISCUSSION THREAD

**Discussion by haimat** [cvat-ai/cvat#8711] [General]
Is there a way to import an annotated dataset into a project directly from a data share mounted on the CVAT server,
or do I have to do so by either uploading local data or importing it from a cloud storage?

**Comment by ariasfx** [comment]
Had the same issue the past week and solved it by running a python script that uploads the data through the CVAT API. 

Maybe the following can help:

```
def upload_images_to_task(task_id, image_paths):
    url = f'{CVAT_API_URL}/tasks/{task_id}/data'
    data = {
        'image_quality': 70
    }
    files = {}
    file_objects = []
    for idx, image_path in enumerate(image_paths):
        try:
            image_file = open(image_path, 'rb')
            file_objects.append(image_file)
            unique_filename = f"{uuid.uuid4()}_{os.path.basename(image_path)}"
            files[f'client_files[{idx}]'] = (unique_filename, image_file, 'image/png')
        except Exception as e:
            logging.exception(f'Failed to open image file {image_path}: {e}')
            continue  # Skip this file and continue with others
    if not files:
        logging.error(f'No valid image files to upload for task {task_id}')
        return
    try:
        response = requests.post(url, headers={'Authorization': f'Token {API_TOKEN}'}, data=data, files=files)
        if response.status_code == 202:
            logging.info(f'{len(image_paths)} images uploaded to task {task_id}')
        else:
            logging.error(f'Failed to upload images to task {task_id}: {response.status_code} {response.text}')
    except Exception as e:
        logging.exception(f'Exception occurred while uploading images to task {task_id}: {e}')
    finally:
        for f in file_objects:
            f.close()
```

image_paths, in this case, would refer to the paths to the images you're looking to upload.

**Reply by haimat** [reply]
@ariasfx Thanks a lot for your code.
But where does it upload the annotations per image?

**Reply by ariasfx** [reply]
@haimat Seems like I completely missed the -annotated- dataset part. Apologies! My use case imports unannotated data for annotation within CVAT, so I've mostly only dealt with the annotation export functionality.

However, you can probably follow a similar methodology to upload annotations by also using doing a POST /api/jobs/{id}/annotations/ request in addition to the image upload. Depending on your operational structure, you might have to go through the tasks endpoint instead.
